Documentation/introduction/detailed_support.rst

```diff
@@ -110,7 +110,7 @@ Moxa NP51x0
 
 A port to the Moxa NP51x0 series of 2-port advanced
 RS-232/422/485 serial device servers was contributed by Anton D.
-Kachalov in NuttX-7.11. This port includes a NuttShell (NSH)
+Kachalov in NuttX-7.11. This port includes :ref:`NSH <nsh>`
```

Comment:
Please have a look here: https://nuttx.apache.org/docs/latest/contributing/documentation.html#linking
My intention was to not have to insert labels everywhere whenever we need to link to some section. So this could be done by linking as <pre>:ref:\`NSH </components/nsh/nsh.rst>`.</pre>

Reply:
But labels are the preferred way otherwise when things move everything has be to updated.  I was actually about to make a comment about how this should be updated.

Reply:
"Using ref is advised over standard reStructuredText links to sections (like `Section title`_) because it works across files, when section headings are changed, and for all builders that support cross-references."
https://www.sphinx-doc.org/en/1.7/markup/inline.html#role-ref

Reply:
Ok, I kind of liked the auto label but I agree it requires more vigilance over not changing section names. 
We should then disable the autosectionlabel extension, update the contributing section as you suggest and convert existing uses of section names as labels. This could be done in a separate PR.

Reply:
before the rst links were to html which is also wrong.  Can I leave this as is here or do you want me to rebase my work on top of yours?

Reply:
Yes, if we will switch autosectionlabel off they way you did it is the way it will be done in general.

Reply:
@v01d @btashton If we switch autosection label off– we'll need to correct all the links in the current Supported Boards area (and everywhere else). I'm wiling to help, let me know.

Reply:
Cool. I'll add a task to #1638 for it to remind us about this.